ports/stm32/mboot/main.c

```diff
@@ -972,6 +1012,11 @@ static int dfu_process_dnload(void) {
             if (dfu_context.wLength == 1) {
                 // mass erase
                 ret = do_mass_erase();
+                if (ret == -1) {
+                    dfu_context.cmd = DFU_CMD_NONE;
+                    return DFU_STATE_ERROR;
```

Comment:
Just noting the early return here is inconsistent with the rest of this function, which uses the `ret` value instead and just lets execution fall to the bottom.